File: packages/shared-types/src/notification.ts
/** Notification item */
export interface NotificationItem {
  notificationId: string;
  title: string;
  message: string;
  type: 'info' | 'warning' | 'error' | 'success';
  isRead: boolean;
  createdAt: string;
  receiverId?: string;
}

File: mohamy-smart-backend/Lawyer.Core/Models/Notification.cs
using Lawyer.Core.Enum;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lawyer.Core.Models
{
    public class Notification : BaseEntity<Guid>
    {
        public Guid ReceiverId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Message { get; set; } = string.Empty;
        public NotificationType Type { get; set; }
        public bool IsRead { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ApplicationUser Receiver { get; set; } = null!;
    }
}
